### PR TITLE
Outline implementation of new demultiplex finish functionality

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR ports part of the required functionality to the temporary `cg demultiplex finish` command, which will replace the existing finish flow cell functionality, and outlines the remaining steps to be taken.

If you feel like you can give me any feedback on whether steps are missing below for the post processing of a flow cell, it is super appreciated!

```
        1. Validate that the flow cell directory exists.
        2. Validate that the demultiplexing data transfer is complete.
        3. Create flow cell.
        4. Store flow cell in status db.
        5. Store flow cell data in housekeeper.
        6. Create sequencing metrics.
```

I'm most uncertain about:

- Is it necessary to check whether delivery already has been started and does this differ for the different sequencing platforms? (Why is there only a specific `is_hiseq_x_delivery_started` function?)
- Where can the sequencer type and name be found? ~Currently that data is retrieved from `cgstats` which will be removed.~

### Added
- Outline for porting functionality to the new `cg demultixplex finish` command 

### How to test
The PR only adds incomplete stubs to the temporary command (`cg demultiplex finish temporary`) which will fail. The intention is to get some feedback on the post processing steps after demultiplexing and outline an implementation plan.

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/meta/demultiplex/demux_post_processing.py [hunk 1]
```diff
@@ -82,7 +82,43 @@ def add_sample_lane_sequencing_metrics_for_flow_cell(self, flow_cell_name: str):
...
+        try:
+            flow_cell: FlowCell = FlowCell(
+                flow_cell_path=flow_cell_dir, bcl_converter=bcl_converter
+            )
+        except FlowCellError:
+            return None
+
+        # 4. Store flow cell in status db.
+        self.status_db.session.add(flow_cell)
+        self.status_db.session.commit()
```
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:117]: I think these are all the steps which are needed indeed. Would be nice to separate them into smaller functions to keep them more testable and simple.

**Comment by ChrOertlin** [comment]
_Where can the sequencer type and name be found? Currently that data is retrieved from cgstats which will be removed._
This should be in the runparameter files, right?

File: cg/meta/demultiplex/demux_post_processing.py (hunk 1 above)
Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:117]: Yes, that sounds like a good idea :)

**Comment by ChrOertlin** [comment]
_Is it necessary to check whether delivery already has been started and does this differ for the different sequencing platforms? (Why is there only a specific is_hiseq_x_delivery_started function?)_

I think we discussed this in the meeting and it was not necessary.

**Comment by diitaz93** [comment]
> _Where can the sequencer type and name be found? Currently that data is retrieved from cgstats which will be removed._ This should be in the runparameter files, right?

It is actually an attribute of the FlowCell object and is retrieved by the name of the flow cell. it is called `sequencer_type`

File: cg/meta/demultiplex/demux_post_processing.py (hunk 1 above)
Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:117]: I'll keep this outline for now and refactor each step in separate PR:s.

**Review by diitaz93 — APPROVED** [review]
looks good 👍

File: cg/meta/demultiplex/demux_post_processing.py [hunk 2]
```diff
@@ -82,7 +82,43 @@ def add_sample_lane_sequencing_metrics_for_flow_cell(self, flow_cell_name: str):
...
+        # 2. Validate that the demultiplexing data transfer is complete.
+        if not Path(flow_cell_dir, DemultiplexingDirsAndFiles.DEMUX_COMPLETE).exists():
+            return
+
+        # 3. Create flow cell. TODO: Populate all fields here.
+        try:
+            flow_cell: FlowCell = FlowCell(
+                flow_cell_path=flow_cell_dir, bcl_converter=bcl_converter
+            )
+        except FlowCellError:
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/meta/demultiplex/demux_post_processing.py:114]: Added some log messages 👍

**Pull request merged**